src/js/components/reply-actions.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import bindAll from 'lodash.bindall';
+
+import { sendMessage, sendLocation } from '../services/conversation';
+import { getRGB, rgbToHsl } from '../utils/colors';
+import { Location } from './location-icon';
+
+export class ReplyActionsComponent extends Component {
+    static propTypes = {
+        accentColor: PropTypes.string,
+        isAccentColorDark: PropTypes.bool,
+        choices: PropTypes.array.isRequired,
+        locationNotSupportedText: PropTypes.string.isRequired
+    };
+
+    constructor(...args) {
+        super(...args);
+        bindAll(this, 'onReplyClick');
+
+        this.state = {};
+    }
+
+    onReplyClick({text}) {
+        const {dispatch} = this.props;
+        dispatch(sendMessage(text));
+    }
+
+    onSendLocation() {
+        const {dispatch, locationNotSupportedText} = this.props;
+
+        if ('geolocation' in navigator) {
+            dispatch(sendLocation());
+        } else {
+            alert(locationNotSupportedText);
```

Comment:
Browser support is broad enough, but we might have problem with people who just disabled the geolocation (and it might not be their fault.. corporate policies or something like that).